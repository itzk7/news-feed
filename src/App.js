import './App.css';
import { Route,   Routes } from 'react-router-dom';
import Navbar from './component/navbar';
import Stories from './component/stories';

function App() {
  return (
    <>
    <Navbar/>
    <Routes>
      <Route exact path = "/" element={Stories} />
      <Route exact path = "/favourties" element={Stories} />
    </Routes>
    </>
  );
}

export default App;
