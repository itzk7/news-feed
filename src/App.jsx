import './App.css';
import { Route,   Routes, BrowserRouter } from 'react-router-dom';
import Navbar from './components/navbar';
import Stories from './components/stories';
import FavoriteStories from './components/favourites';
import { FavoriteIDsContext } from './components/context';
import { useState } from 'react';

function App() {
  const [favIds, setFavIds] = useState([])
  return (
    <>
    <FavoriteIDsContext.Provider value={[favIds, setFavIds]}>
          <BrowserRouter>
          <Navbar/>
            <div className='main'>
              <Routes>
                  <Route exact path="/" element={<Stories />} />
                  <Route exact path="/favorites" element={<FavoriteStories />} />
              </Routes>
            </div>
          </BrowserRouter>
    </FavoriteIDsContext.Provider>
    </>
  );

}

export default App;
