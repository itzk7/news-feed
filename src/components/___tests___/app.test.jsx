import React from "react";
import App from "../../App";
import {setupServer} from 'msw/node'
import {rest} from 'msw'
import { render, screen } from "@testing-library/react";

const server = setupServer(
    rest.get('https://hacker-news.firebaseio.com/v0/item/123.json', async (req, res, ctx) => {
    const story = {
        title: "Story 1",
        by: "Steve Rogers",
        descendants: 100,
        time: 1691309212,
        id: 123
    }
      return res(ctx.json(story))
    }),
    rest.get('https://hacker-news.firebaseio.com/v0/topstories.json', async (req, res, ctx) => {
    return res(ctx.json([123]))
  }),
)

beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())


describe("Jest Snapshot testing suite",  () => {
  it("Matches DOM Snapshot", async () => {
    const {asFragment} = render(<App/>);
    await screen.findByText(/Steve Rogers/)
    
    expect(asFragment()).toMatchSnapshot();
  });
});
