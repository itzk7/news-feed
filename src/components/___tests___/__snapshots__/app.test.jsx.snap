// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <div
      class="container"
    >
      <div
        class="title-container"
      >
        <svg>
          batman.svg
        </svg>
        <h2>
          Wayne enterprise
        </h2>
      </div>
      <div
        class="menu-icon"
      >
        <svg>
          hamburger.svg
        </svg>
      </div>
      <div
        class="nav-elements  false"
      >
        <ul>
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Stories
            </a>
          </li>
          <li>
            <a
              class=""
              href="/favorites"
            >
              Favourites (0)
            </a>
          </li>
          <li>
            <a
              class=""
              href="/about"
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    class="main"
  >
    <div
      class="stories-container"
    >
      <h1>
        Top stories
      </h1>
    </div>
    <div
      class="stories-container"
    >
      <div
        class="story-container"
      >
        <div
          class="story-content"
        >
          <a
            class="story-title"
            rel="noreferrer"
            target="_blank"
          >
            Story 1
          </a>
          <p>
            - By,  Steve Rogers 
          </p>
          <div
            class="story-meta"
          >
            <p>
              ðŸ—“ August 6th 2023, ðŸ•’ 1:36:52 pm
            </p>
            <p>
              ðŸ’¬ Comments (100)
            </p>
            <button
              class="icon-button"
            >
              <svg>
                fav-not-selected.svg
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
